src/picom.c

```diff
@@ -280,6 +280,37 @@ static bool run_fade(session_t *ps, struct managed_win **_w, long steps) {
 	return true;
 }
 
+// === Animations ===
+
+/**
+ * Get the time delta of a window animation.
+ *
+ * In milliseconds.
+ */
+static long animation_delta(session_t *ps) {
+	long delta = ps->o.animation_duration >> 7; // /128
+	delta = max2(delta, 1);
+	return delta;
+}
+
+/**
+ * Get the time left before next window animation point.
+ *
+ * In milliseconds.
+ */
+static double animation_timeout(session_t *ps) {
+	long double now = get_time_ms();
```

Comment:
In #766 , @Arian8j2 raised concerns about using wall-clock time for animation. Do you think it would be fine?

Reply:
I think it should be fine -- it appears to be accurate enough, and, in the case of innacuracies, the math for the animation should be robust in that the window will converge to it's final location and overly weird things shouldn't happen. I haven't noticed any issues at all, nor were there any issues reported which had any relation to the fact that we use wall clock time.

On the other hand, I couldn't find any notes by @Arian8j2 in https://github.com/yshui/picom/pull/766 regarding use of wall clock. Where can I find it?

Reply:
Here: https://github.com/yshui/picom/pull/766#discussion_r794391499

Reply:
At first of writing transitions i tried to use delta time for calculating animations like anywhere else, but i just seen a huge difference in some frame times and in that point i just gave up and sticked to calculating for 60 fps and latter i changed it to be based on refresh rate but then @yshui mentioned using steps just like fading step i changed the code to work with steps. Maybe using wall-clock was ok and i was just nitpicking, I should take a second look to it.